packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/function.js

```diff
@@ -26,5 +29,25 @@ export const visit_function = (node, context) => {
 		});
 	}
 
-	context.next(state);
+	if (in_constructor && (state.private_state.size > 0 || state.public_state.size > 0)) {
+		const params = /** @type {Pattern[]} */ (node.params.map((p) => context.visit(p, state)));
+		const body = /** @type {BlockStatement} */ (context.visit(node.body, state));
+		const has_super_call =
+			body.body[0]?.type === 'ExpressionStatement' &&
+			body.body[0].expression.type === 'CallExpression' &&
+			body.body[0].expression.callee.type === 'Super';
```

Comment:
the `super()` call doesn't need to be the first thing in the constructor. it doesn't even need to be at the top level